src/lib.rs

```diff
@@ -89,12 +127,10 @@ pub trait GpioOut {
     }
 
     /// Set the GPIO port to a low output value directly
-    #[inline(always)]
-    fn set_low(&mut self) -> Result<(), Self::Error>;
+    fn set_low(&self) -> Result<(), Self::Error>;
```

Comment:
I am not sure I like having write operations being possible using a `&` instead of `&mut`. While it may be feasible in the sysfs case due to the Linux kernel multiplexing, does it really seem intuitive on a larger scale?

For one, when using bare memory GPIO access, this will now require an additional interior mutability struct (otherwise, the memory could just be read/written). Additionally, the semantics are unclear: If a single GPIO is more like a "packet", that is fully self-contained because the other end is just waiting on edges, it would probably be good. I do think that a continuous output stream might also be a use-case though?